lib/streamlit/server/Server.py

```diff
@@ -116,8 +118,35 @@ def start(self, on_started):
 
         LOGGER.debug("Starting server...")
         app = self._create_app()
+
+        class PortAlradyInUse(Exception):
+            pass
+
+        def start_listening(rotate=True):
+            port = config.get_option("server.port")
+            try:
+                app.listen(port)
+            except OSError as e:
+                ADDRESS_ALREADY_IN_USE = 48
+                if e.errno == ADDRESS_ALREADY_IN_USE:
+                    if config.is_manually_set("server.port"):
+                        LOGGER.debug(
+                            "Port %s already in use, trying next available one", port
+                        )
+                        port += 1
```

Comment:
Should we add boundaries for this?, just to be sure

Reply:
boundaries like "try only 100 times" or "don't get past port 8000"?

Reply:
I think a limit in the attempts would be wise to avoid some sort of infinite loop

Reply:
Exactly that :)

Reply:
added a limit of 100 simultaneous Streamlits